### PR TITLE
Restore folder for CLI (AWS support only)

### PR DESCRIPTION
**Pull request by ekazachkova**
This PR provides implementation for folder restore for Cloud Pipeline CLI (#792). For now this operation supports AWS provider only.

### CODE REVIEW

**Comment by sidoruka** [comment]
@NShaforostov please take a look as well:
* Test it
* Update the CLI docs (shall be added into this PR)

File: pipe-cli/src/utilities/storage/s3.py [hunk 1]
```diff
@@ -300,6 +303,30 @@ def load_delete_marker(self, bucket, path, client):
...
+        if path:
+            operation_parameters['Prefix'] = prefix
+        paginator = client.get_paginator('list_object_versions')
+        pages = paginator.paginate(**operation_parameters)
+        restore_us = dict(Objects=[])
+        for page in pages:
+            S3BucketOperations.process_listing(page, 'DeleteMarkers', restore_us, delimiter, exclude, include, prefix,
+                                               versions=True)
+            # flush once aws limit reached
+            if len(restore_us['Objects']) >= 1000:
```
Comment (mzueva) [inline pipe-cli/src/utilities/storage/s3.py]: It seems, that this code will fail if `len(restore_us['Objects']) > 1000`

**Review by mzueva — CHANGES_REQUESTED** [review]

File: pipe-cli/src/utilities/storage/s3.py (hunk 1 above)
Comment (ekazachkova) [reply pipe-cli/src/utilities/storage/s3.py]: fixed for deletion too

**Review by mzueva — APPROVED** [review]

**Comment by mzueva** [comment]
@ekazachkova I've tested this branch and it seems to work fine, except for the following case: if I try to restore a file with `-r` option, nothing happens. @NShaforostov @sidoruka we need to decide, how to handle this case - throw an error if restored object is not a folder or restore it as file.

**Comment by sidoruka** [comment]
> @ekazachkova I've tested this branch and it seems to work fine, except for the following case: if I try to restore a file with `-r` option, noting happens. @NShaforostov @sidoruka we need to decide, how to handle this case - throw an error if restored object is not a folder or restore it is file.

@mzueva how do we handle it for `pipe cp`?

**Comment by NShaforostov** [comment]
Verified, works fine!

**Comment by NShaforostov** [comment]
@sidoruka the CLI docs updated via [be7ad17](https://github.com/epam/cloud-pipeline/pull/789/commits/be7ad1772be461c2962ba843bf2376cd255c0e4c).

**Pull request merged**